docs/naming_conventions.md

```diff
@@ -0,0 +1,150 @@
+
+# Naming Conventions in Kids First FHIR Data Model
+
+This document describes the naming conventions for ids and
+the file names for resources in the Kids First FHIR data model.
+
+## Resource Id
+
+### Semantics
+
+The `id` attribute for a resource in the FHIR model should be
+populated with some kind of logical value that describes the semantics of
+the resource (e.g. `ResearchParticipant` vs `2349703`).
+
+**NOTE:** This ONLY applies to the files in the FHIR model. Non-conformance
+resources (representing real data) loaded into the FHIR server will likely
+have an autogenerated alphanumeric ID which may or may not be random.
+
+### Syntax
+
+The `id` value should conform to the the regex pattern in the
+[FHIR spec](https://www.hl7.org/fhir/datatypes.html#id).
+
+### Casing
+
+All resource `id` values should have the same casing. There does not seem
+to be a consistent standard or recommendation for this but we should pick
+one and stick to it.
+
+Based on observations, the FHIR spec seems to use PascalCase for
+resource types (e.g. `Patient`, `SearchParameter`) and camelCase
+(e.g. `gender`, `birthDate`) for resource attributes
+
+The Kids First model will use [kebab-case](https://stackoverflow.com/questions/11273282/whats-the-name-for-hyphen-separated-case)
+for all resource `id` values. Kebab case is essentially snake case with hyphens
+instead of underscores (e.g. `age-at-event`, `sequencing-experiment`).
+
+
+## Resource File Name
+
+### Conformance Resources
+
+All conformance resource file names should follow this pattern:
+
+**Pattern**
+```
+<resource type>_<base definition resource id>_<resource id>.json
```

Comment:
What about conformance resources that do not have separate base definitions such as CodeSystem, ValueSet, and SearchParameter? Then, `CodeSystem_CodeSystem_use-core-ethnicity.json`? Or, should we change this patter `<resource type>_<base definition resource id>_<resource id>.json` to like `<resource type>_[<base definition resource id>_]<resource id>.json` in order to indicate the middle part can be optional?

Reply:
Going with the latter 👍